src/detectors.rs

```diff
@@ -0,0 +1,340 @@
+//! Utility to help with "really nice to add a warning for tasks that might be blocking"
+use libc;
+use rand::thread_rng;
+use rand::Rng;
+use std::collections::HashSet;
+use std::sync::mpsc;
+use std::sync::{Arc, Mutex};
+use std::thread::ThreadId;
+use std::time::Duration;
+use std::{env, thread};
+use tokio::runtime::{Builder, Runtime};
+
+const PANIC_WORKER_BLOCK_DURATION_DEFAULT: Duration = Duration::from_secs(60);
+
+fn get_panic_worker_block_duration() -> Duration {
+    let duration_str = env::var("MY_DURATION_ENV").unwrap_or_else(|_| "60".to_string());
+    duration_str
+        .parse::<u64>()
+        .map(Duration::from_secs)
+        .unwrap_or(PANIC_WORKER_BLOCK_DURATION_DEFAULT)
+}
+
+#[cfg(unix)]
+fn get_thread_id() -> libc::pthread_t {
+    unsafe { libc::pthread_self() }
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub struct ThreadInfo {
+    id: ThreadId,
+    #[cfg(unix)]
+    pthread_id: libc::pthread_t,
+}
+
+impl ThreadInfo {
+    fn new() -> Self {
+        ThreadInfo {
+            id: thread::current().id(),
+            #[cfg(unix)]
+            pthread_id: get_thread_id(),
+        }
+    }
+
+    // Getter for the id field
+    pub fn id(&self) -> &ThreadId {
+        &self.id
+    }
+
+    // Getter for the pthread_id field (only available on Unix)
+    #[cfg(unix)]
```

Comment:
You can use `cfg_attr` to have "only available on unix" show up in the documentation.
```suggestion
    /// Getter for the `pthread_id`.
    #[cfg(unix)]
    #[cfg_attr(docsrs, cfg(unix))]
```